tests/integ/test_dataframe.py

```diff
@@ -2394,6 +2396,84 @@ def test_save_as_table_with_table_sproc_output(session, save_mode, table_type):
         Utils.drop_procedure(session, f"{temp_sp_name}()")
 
 
+@pytest.mark.parametrize("table_type", ["", "temp", "temporary", "transient"])
+@pytest.mark.parametrize("save_mode", ["append", "overwrite"])
+def test_write_table_with_clustering_keys(session, save_mode, table_type):
+    table_name = Utils.random_name_for_temp_object(TempObjectType.TABLE)
+    df1 = session.create_dataframe(
+        [],
+        schema=StructType(
+            [
+                StructField("c1", DateType()),
+                StructField("c2", StringType()),
+                StructField("c3", IntegerType()),
+            ]
+        ),
+    )
+    df2 = session.create_dataframe(
+        [],
+        schema=StructType(
+            [
+                StructField("c1", TimestampType()),
+                StructField("c2", StringType()),
+                StructField("c3", IntegerType()),
+            ]
+        ),
+    )
+    df3 = session.create_dataframe(
+        [],
+        schema=StructType(
+            [StructField("t", TimestampType()), StructField("v", VariantType())]
+        ),
+    )
+    try:
+        with mock.patch(
+            "snowflake.snowpark._internal.server_connection.ServerConnection.execute"
+        ) as execute_mocked:
```

Comment:
Since this is an integration test, I wonder if we could actually create the table and check `describe` or `get_ddl` to see if the created table has `cluster by`. What do you think?

Reply:
Good suggestion. I'll work on it.

Reply:
done